ipam/client/backends/phpipam.py

```diff
@@ -131,6 +144,91 @@ def get_allocated_ips_by_subnet_id(self, subnetid):
         iplist = [IPAddress(int(ip[0])) for ip in self.cur]
         return iplist
 
+    def add_next_subnet(self, parent_subnet, prefixlen, description):
+        if prefixlen <= parent_subnet.prefixlen:
+            raise ValueError('Parent subnet {} is too small to add new '
+                             'subnet with prefixlen {}!'
+                             ''.format(parent_subnet, prefixlen))
+        subnet = self._get_next_free_subnet(parent_subnet, prefixlen)
+        if not subnet:
+            raise ValueError('No more space to add a new subnet with '
+                             'prefixlen {} in {}!'.format(
+                                prefixlen, parent_subnet))
+
+        parent_subnet_id = self.find_subnet_id(parent_subnet)
+        parent_subnet_used_ips = self.get_allocated_ips_by_subnet_id(
+            parent_subnet_id)
+
+        if len(parent_subnet_used_ips):
+            raise ValueError('Parent subnet {} must not contain any '
+                             'allocated IP address!'.format(parent_subnet))
+
+        self.cur.execute(
+            'INSERT INTO subnets '
+            '(subnet, mask, sectionId, description, vrfId, '
+            'masterSubnetId, vlanId, permissions) '
+            'VALUES (\'{:d}\', \'{}\', \'{}\', \'{}\', '
+            '\'{}\', \'{}\', \'{}\', \'{}\')'.format(
+                int(subnet.ip),
+                subnet.prefixlen,
+                self.section_id,
+                description,
+                self.subnet_options['vrf_id'],
+                parent_subnet_id,
+                self.subnet_options['vlan_id'],
+                self.subnet_options['permissions']))
+        return subnet
+
+    def _get_next_free_subnet(self, subnet, prefixlen):
+        """
+        Find next free prefixlen-wide subnet in given subnet.
+        """
+        subnet_id = self.find_subnet_id(subnet)
+        if not subnet_id:
+            raise ValueError('Unable to get subnet id from database '
+                             'for subnet {}'.format(subnet))
+
+        unavailable_subnets = self._get_unavailable_subnets(
+            subnet_id, prefixlen)
+
+        for candidate_subnet in subnet.subnet(prefixlen):
+            if candidate_subnet not in unavailable_subnets:
```

Comment:
Is it mandatory to compare using the same masks ? Don't we have any tooling in netaddr to compare sets (intersections) ?

Reply:
Fixed by switching to ipaddress which provides an overlaps method.